js/validation.js

```diff
@@ -0,0 +1,65 @@
+import {formElement, HOUSING_MIN_PRICE} from './data.js';
+const adTitleInput = formElement.querySelector('#title');
+const typeSelect = formElement.querySelector('#type');
+const priceInput = formElement.querySelector('#price');
+const countRoomsSelect = formElement.querySelector('#room_number');
+const countGuestsSelect = formElement.querySelector('#capacity');
+
+const valuePriceCheck = () => {
+  priceInput.min = HOUSING_MIN_PRICE[typeSelect.value];
+  (Number(priceInput.value) < Number(priceInput.min)) ? priceInput.setCustomValidity(`Минимальная цена за ночь ${  priceInput.min}`) : priceInput.setCustomValidity('');
+  priceInput.reportValidity();
+};
+
+adTitleInput.addEventListener('input', () => {
+  const valueLength = adTitleInput.value.length;
+  const valueMin = adTitleInput.attributes.minlength.value;
+  const valueMax = adTitleInput.attributes.maxlength.value;
+  if (valueLength < valueMin) {
+    adTitleInput.setCustomValidity(`Ещё ${valueMin - valueLength} симв.`);
+  }
+  else if (valueLength > valueMax) {
+    adTitleInput.setCustomValidity(`Удалите лишние ${valueLength - valueMax} симв.`);
+  }
+  else {
+    adTitleInput.setCustomValidity('');
+  }
+  adTitleInput.reportValidity();
+});
+
+typeSelect.addEventListener('change', () => {
+  priceInput.placeholder = String(HOUSING_MIN_PRICE[typeSelect.value]);
```

Comment:
Если убрать конструктор String, то ничего не изменится
```suggestion
  priceInput.placeholder = HOUSING_MIN_PRICE[typeSelect.value];
```

Вообще реальная нужда использовать конструкторы(Number, String, Boolean) для преобразования типов возникает редко, достаточно использовать toString, parseInt, а к Boolean javascript умеет преобразовывать и без дополнительной помощи, т.к. является языком с динамическим приведением типов, но если нужно, то можно воспользоваться двумя знаками восклицания `!!` перед выражением